runtime/service/src/main/java/org/apache/polaris/service/admin/PolarisServiceImpl.java

```diff
@@ -137,13 +137,15 @@ public Response createCatalog(
   }
 
   private void validateClientId(String clientId) {
-    if (!clientId.matches("^[0-9a-f]{16}$")) {
+    if (!clientId.matches(
+        realmConfig.getConfig(FeatureConfiguration.CREDENTIAL_RESET_CLIENT_ID_PATTERN))) {
```

Comment:
we should validate the value of `CREDENTIAL_RESET_CLIENT_ID_PATTERN` on startup to avoid deferred RegEx syntax errors (it's an admin mistake, not the API client's mistake)

Reply:
Cf. https://github.com/apache/polaris/blob/1aa95deb75b9f16c871fb8140576d7026471b2b1/runtime/service/src/main/java/org/apache/polaris/service/config/ProductionReadinessChecks.java#L142